libs/domain/cart/edit/edit.component.ts

```diff
@@ -111,30 +110,38 @@ export class CartEditComponent extends CartComponentMixin(
     const fields = this.getFields();
     const values = this.$values();
 
-    return html`<form @submit=${this.onSubmit}>
+    return html`<form @oryx.submit=${this.onSubmit}>
       <oryx-layout
         layout="grid"
         style="--oryx-column-count:2;--column-gap: 20px;"
       >
         ${this.fieldRenderer.buildForm(fields, values)}
 
-        <oryx-button @click=${() => this.submit()}
+        <oryx-button ?loading=${this.isLoading} @click=${() => this.submit()}
           >${this.i18n('create')}</oryx-button
         >
       </oryx-layout>
     </form>`;
   }
 
-  protected onSubmit(e: SubmitEvent): void {
-    e.preventDefault();
-
-    const form = e.target as HTMLFormElement;
-
-    const cart = Object.fromEntries(new FormData(form).entries());
-    cart.store = this.$store()!.id;
-
-    //TODO: replace with cart create service call
-    //implemented in list story
-    console.log(cart);
+  protected onSubmit(e: CustomEvent): void {
+    this.isLoading = true;
+
+    this.cartService.createCart(e.detail.values).subscribe({
+      next: (cart) => {
+        this.notificationService.push({
+          type: AlertType.Success,
+          //TODO: use translation string when issue with resolving of translated result
+          //outside components render is fixed
+          // content: this.i18n('carts.create.cart-<name>-created', { name: cart.name }) as string
+          content: `Cart: ${cart.name} was created`,
+        });
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.routerService.navigate(this.$redirectLink()!);
+      },
+      complete: () => {
+        this.isLoading = false;
```

Comment:
Do we cover error?

Reply:
will be handled by our global error handler (show the notification)